user-guide/modules/ROOT/pages/libraries-that-work-together.adoc

```diff
@@ -0,0 +1,32 @@
+= Boost Libraries that Work Well Together
+:navtitle: Libraries that Work Well Together
+
+Boost libraries each has its own use, but many are designed to work together to simplify complex tasks. Here are a few examples of Boost libraries that work well together:
+
+== Working Well in Pairs
```

Comment:
boost.json and boost.describe seem like a trendy combination lately.

Reply:
OK - I will add that as a pair.

Reply:
And is the combined purpose to use Boost.Describe to automatically generate code to serialize and deserialize custom data types to and from JSON? Or is there another reason for the combo?

Reply:
Yes. I think that's it.

Reply:
Describe is meant to add reflection to stuff. A useful consequence of it is being able to serialize or deserialize to JSON. But you can use it to add operators like `operator==` and `operator<<` (really useful in unit tests). There is a (almost finished) Boost.MySQL feature that uses Describe as an interface for custom row types. So the end user creates a struct annotated with Boost.Describe macros, and Boost.MySQL is automatically able to parse the results of a `SELECT` query into that type. You can use it to convert a struct into a tuple, too. In general, anything using `std::tuple` could instead use a Boost.Describe struct.

Reply:
@anarthal Good info, we can add the SQL example when the lib is released.

Reply:
Boost.MySQL and Describe is a good idea too. I didn't know about this feature. That's great. 

I think Boost.Json and Describe came to mind because I remember them saying they were working on that. But I don't remember if that's a feature in Boost.Describe or Boost.Json.